krm-functions/lib/condkptsdk/sdk_updates.go

```diff
@@ -0,0 +1,170 @@
+		}
+		r.kptf.SetConditions(c)
+	} else {
+		objRef := refs[1]
+		c := kptv1.Condition{
+			Type:    kptfilelibv1.GetConditionType(objRef),
+			Status:  status,
+			Reason:  fmt.Sprintf("%s.%s", kptfilelibv1.GetConditionType(&r.cfg.For), forRef.Name),
+			Message: fmt.Sprintf("%s %s", a, msg),
+		}
+		r.kptf.SetConditions(c)
+		// update the condition status back in the inventory
+		if err := r.inv.set(&gvkKindCtx{gvkKind: kind}, []corev1.ObjectReference{*forRef, *objRef}, &c, false); err != nil {
+			fn.Logf("error updating stage1 resource to the inventory: %v\n", err.Error())
+			r.rl.Results = append(r.rl.Results, fn.ErrorResult(err))
+		}
+	}
+}
+
+func (r *sdk) setObjectInResourceList(kind gvkKind, refs []*corev1.ObjectReference, obj *object) {
+	if !IsRefsValid(refs) {
+		return
+	}
+	forRef := refs[0]
+	if len(refs) == 1 {
+		if err := r.rl.UpsertObjectToItems(&obj.obj, nil, true); err != nil {
+			fn.Logf("error updating stage1 resource to the inventory: %v\n", err.Error())
+			r.rl.Results = append(r.rl.Results, fn.ErrorResult(err))
+		}
+		// update the resource status back in the inventory
+		if err := r.inv.set(&gvkKindCtx{gvkKind: kind}, []corev1.ObjectReference{*forRef}, &obj.obj, false); err != nil {
+			fn.Logf("error updating stage1 resource to the inventory: %v\n", err.Error())
+			r.rl.Results = append(r.rl.Results, fn.ErrorResult(err))
+		}
+	} else {
+		objRef := refs[1]
+		//r.rl.SetObject(&obj.obj)
+		if err := r.rl.UpsertObjectToItems(&obj.obj, nil, true); err != nil {
+			fn.Logf("error updating stage1 resource: %v\n", err.Error())
+			r.rl.Results = append(r.rl.Results, fn.ErrorResult(err))
+		}
+		// update the resource status back in the inventory
+		if err := r.inv.set(&gvkKindCtx{gvkKind: kind}, []corev1.ObjectReference{*forRef, *objRef}, &obj.obj, false); err != nil {
+			fn.Logf("error updating stage1 resource to the inventory: %v\n", err.Error())
+			r.rl.Results = append(r.rl.Results, fn.ErrorResult(err))
+		}
+	}
+}
+
+func IsRefsValid(refs []*corev1.ObjectReference) bool {
+	if len(refs) == 0 || (len(refs) == 1 && refs[0] == nil) || (len(refs) == 2 && refs[0] == nil && refs[1] == nil) {
+		return false
+	}
+	return true
+}
+
+func (r *sdk) updateKptFile() {
+	kptfile, err := r.kptf.ParseKubeObject()
+	if err != nil {
+		fn.Log(err)
+		r.rl.Results = append(r.rl.Results, fn.ErrorConfigObjectResult(err, r.rl.Items.GetRootKptfile()))
+	}
+	if err := r.rl.UpsertObjectToItems(kptfile, nil, true); err != nil {
+		fn.Log(err)
+		r.rl.Results = append(r.rl.Results, fn.ErrorConfigObjectResult(err, r.rl.Items.GetRootKptfile()))
+	}
+}
+
+func (r *sdk) deleteObjFromResourceList(obj *fn.KubeObject) {
+	for idx, o := range r.rl.Items {
+		if isGVKNNEqual(o, obj) {
+			r.rl.Items = append(r.rl.Items[:idx], r.rl.Items[idx+1:]...)
+		}
+	}
+}
+
+// isGVKNEqual validates if the APIVersion, Kind, Name and Namespace of both fn.KubeObject are equal
+func isGVKNNEqual(curobj, newobj *fn.KubeObject) bool {
```

Comment:
Probably a good candidate for a base library

Reply:
I added in utils for now